Missing default ctor for HazelcastJsonValue

C++ compiler version: gcc 9.3
Hazelcast Cpp client version: 3.12
Hazelcast server version: 3.12
Number of the clients: 1
Cluster size, i.e. the number of Hazelcast cluster members: 3
OS version (Windows/Linux/OSX): Linux Ubuntu 20.04

#### Expected behaviour
`.values()` method for `IMap<std::string, HazelcastJsonValue>` with no arguments compiles successfully and produce correct output(all values in the map)

#### Actual behaviour
```stacktrace
/usr/include/c++/9/bits/stl_construct.h: In instantiation of ‘void std::_Construct(_T1*, _Args&& ...) [with _T1 = hazelcast::client::HazelcastJsonValue; _Args = {}]’:
/usr/include/c++/9/bits/stl_uninitialized.h:545:18:   required from ‘static _ForwardIterator std::__uninitialized_default_n_1<_TrivialValueType>::__uninit_default_n(_ForwardIterator, _Size) [with _ForwardIterator = hazelcast::client::HazelcastJsonValue*; _Size = long unsigned int; bool _TrivialValueType = false]’
/usr/include/c++/9/bits/stl_uninitialized.h:601:20:   required from ‘_ForwardIterator std::__uninitialized_default_n(_ForwardIterator, _Size) [with _ForwardIterator = hazelcast::client::HazelcastJsonValue*; _Size = long unsigned int]’
/usr/include/c++/9/bits/stl_uninitialized.h:663:44:   required from ‘_ForwardIterator std::__uninitialized_default_n_a(_ForwardIterator, _Size, std::allocator<_Tp>&) [with _ForwardIterator = hazelcast::client::HazelcastJsonValue*; _Size = long unsigned int; _Tp = hazelcast::client::HazelcastJsonValue]’
/usr/include/c++/9/bits/stl_vector.h:1603:36:   required from ‘void std::vector<_Tp, _Alloc>::_M_default_initialize(std::vector<_Tp, _Alloc>::size_type) [with _Tp = hazelcast::client::HazelcastJsonValue; _Alloc = std::allocator<hazelcast::client::HazelcastJsonValue>; std::vector<_Tp, _Alloc>::size_type = long unsigned int]’
/usr/include/c++/9/bits/stl_vector.h:509:9:   required from ‘std::vector<_Tp, _Alloc>::vector(std::vector<_Tp, _Alloc>::size_type, const allocator_type&) [with _Tp = hazelcast::client::HazelcastJsonValue; _Alloc = std::allocator<hazelcast::client::HazelcastJsonValue>; std::vector<_Tp, _Alloc>::size_type = long unsigned int; std::vector<_Tp, _Alloc>::allocator_type = std::allocator<hazelcast::client::HazelcastJsonValue>]’
/home/rikevoltz/.conan/data/hazelcast-cpp-client/3.12/_/_/package/49e7f59961e8ed9b7d1d33caa2e6d613b00b72a5/include/hazelcast/client/map/ClientMapProxy.h:751:36:   required from ‘std::vector<V> hazelcast::client::map::ClientMapProxy<K, V>::values() [with K = std::__cxx11::basic_string<char>; V = hazelcast::client::HazelcastJsonValue]’
/home/rikevoltz/.conan/data/hazelcast-cpp-client/3.12/_/_/package/49e7f59961e8ed9b7d1d33caa2e6d613b00b72a5/include/hazelcast/client/IMap.h:603:40:   required from ‘std::vector<V> hazelcast::client::IMap<K, V>::values() [with K = std::__cxx11::basic_string<char>; V = hazelcast::client::HazelcastJsonValue]’
/home/rikevoltz/work/Projects/hazelcast_tests/main.cpp:22:34:   required from here
/usr/include/c++/9/bits/stl_construct.h:75:7: error: no matching function for call to ‘hazelcast::client::HazelcastJsonValue::HazelcastJsonValue()’
   75 |     { ::new(static_cast<void*>(__p)) _T1(std::forward<_Args>(__args)...); }
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/rikevoltz/work/Projects/hazelcast_tests/main.cpp:2:
/home/rikevoltz/.conan/data/hazelcast-cpp-client/3.12/_/_/package/49e7f59961e8ed9b7d1d33caa2e6d613b00b72a5/include/hazelcast/client/HazelcastJsonValue.h:58:13: note: candidate: ‘hazelcast::client::HazelcastJsonValue::HazelcastJsonValue(const string&)’
   58 |             HazelcastJsonValue(const std::string &jsonString);
      |             ^~~~~~~~~~~~~~~~~~
/home/rikevoltz/.conan/data/hazelcast-cpp-client/3.12/_/_/package/49e7f59961e8ed9b7d1d33caa2e6d613b00b72a5/include/hazelcast/client/HazelcastJsonValue.h:58:13: note:   candidate expects 1 argument, 0 provided
/home/rikevoltz/.conan/data/hazelcast-cpp-client/3.12/_/_/package/49e7f59961e8ed9b7d1d33caa2e6d613b00b72a5/include/hazelcast/client/HazelcastJsonValue.h:48:29: note: candidate: ‘hazelcast::client::HazelcastJsonValue::HazelcastJsonValue(const hazelcast::client::HazelcastJsonValue&)’
   48 |         class HAZELCAST_API HazelcastJsonValue {
      |                             ^~~~~~~~~~~~~~~~~~
/home/rikevoltz/.conan/data/hazelcast-cpp-client/3.12/_/_/package/49e7f59961e8ed9b7d1d33caa2e6d613b00b72a5/include/hazelcast/client/HazelcastJsonValue.h:48:29: note:   candidate expects 1 argument, 0 provided
```
As I understand correctly, the default ctor for HazelcastJsonValue is missing
#### Steps to reproduce the behaviour
```c++
  auto client = hazelcast::client::HazelcastClient{config};
  auto some_map = client.getMap<std::string, HazelcastJsonValue>("test_map");
  some_map.put("some_key", HazelcastJsonValue("{\"id\":\"some_id\"}"));
  some_map.put("some_other_key", HazelcastJsonValue("{\"id\":\"some_other_id\"}"));
  some_map.put("another_key", HazelcastJsonValue("{\"id\":\"another_id\"}"));
  auto results = some_map.values();
```

--- a/hazelcast/include/hazelcast/client/HazelcastJsonValue.h
+++ b/hazelcast/include/hazelcast/client/HazelcastJsonValue.h
@@ -69,6 +69,8 @@ namespace hazelcast {
 
             bool operator!=(const HazelcastJsonValue &rhs) const;
 
+            bool operator<(const HazelcastJsonValue &rhs) const;
+
             friend std::ostream HAZELCAST_API &operator<<(std::ostream &os, const HazelcastJsonValue &value);
 
         private:

--- a/hazelcast/include/hazelcast/client/map/ClientMapProxy.h
+++ b/hazelcast/include/hazelcast/client/map/ClientMapProxy.h
@@ -654,10 +654,11 @@ namespace hazelcast {
                 std::vector<K> keySet() {
                     std::vector<serialization::pimpl::Data> dataResult = proxy::IMapImpl::keySetData();
                     size_t size = dataResult.size();
-                    std::vector<K> keys(size);
+                    std::vector<K> keys;
+                    keys.reserve(size);
                     for (size_t i = 0; i < size; ++i) {
                         std::unique_ptr<K> key = toObject<K>(dataResult[i]);
-                        keys[i] = *key;
+                        keys.push_back(*key);
                     }
                     return keys;
                 }
@@ -693,10 +694,11 @@ namespace hazelcast {
                 std::vector<K> keySet(const query::Predicate &predicate) {
                     std::vector<serialization::pimpl::Data> dataResult = proxy::IMapImpl::keySetData(predicate);
                     size_t size = dataResult.size();
-                    std::vector<K> keys(size);
+                    std::vector<K> keys;
+                    keys.reserve(size);
                     for (size_t i = 0; i < size; ++i) {
                         std::unique_ptr<K> key = toObject<K>(dataResult[i]);
-                        keys[i] = *key;
+                        keys.push_back(*key);
                     }
                     return keys;
                 }
@@ -747,10 +749,11 @@ namespace hazelcast {
                 std::vector<V> values() {
                     std::vector<serialization::pimpl::Data> dataResult = proxy::IMapImpl::valuesData();
                     size_t size = dataResult.size();
-                    std::vector<V> values(size);
+                    std::vector<V> values;
+                    values.reserve(size);
                     for (size_t i = 0; i < size; ++i) {
                         std::unique_ptr<V> value = toObject<V>(dataResult[i]);
-                        values[i] = *value;
+                        values.push_back(*value);
                     }
                     return values;
                 }
@@ -782,6 +785,7 @@ namespace hazelcast {
                     std::vector<serialization::pimpl::Data> dataResult = proxy::IMapImpl::valuesData(predicate);
                     size_t size = dataResult.size();
                     std::vector<V> values;
+                    values.reserve(size);
                     for (size_t i = 0; i < size; ++i) {
                         std::unique_ptr<V> value = toObject<V>(dataResult[i]);
                         values.push_back(*value);
@@ -826,11 +830,12 @@ namespace hazelcast {
                 std::vector<std::pair<K, V> > entrySet() {
                     std::vector<std::pair<serialization::pimpl::Data, serialization::pimpl::Data> > dataResult = proxy::IMapImpl::entrySetData();
                     size_t size = dataResult.size();
-                    std::vector<std::pair<K, V> > entries(size);
+                    std::vector<std::pair<K, V> > entries;
+                    entries.reserve(size);
                     for (size_t i = 0; i < size; ++i) {
                         std::unique_ptr<K> key = toObject<K>(dataResult[i].first);
                         std::unique_ptr<V> value = toObject<V>(dataResult[i].second);
-                        entries[i] = std::make_pair(*key, *value);
+                        entries.emplace_back(*key, *value);
                     }
                     return entries;
                 }
@@ -851,11 +856,12 @@ namespace hazelcast {
                     std::vector<std::pair<serialization::pimpl::Data, serialization::pimpl::Data> > dataResult = proxy::IMapImpl::entrySetData(
                             predicate);
                     size_t size = dataResult.size();
-                    std::vector<std::pair<K, V> > entries(size);
+                    std::vector<std::pair<K, V> > entries;
+                    entries.reserve(size);
                     for (size_t i = 0; i < size; ++i) {
                         std::unique_ptr<K> key = toObject<K>(dataResult[i].first);
                         std::unique_ptr<V> value = toObject<V>(dataResult[i].second);
-                        entries[i] = std::make_pair(*key, *value);
+                        entries.emplace_back(*key, *value);
                     }
                     return entries;
                 }
@@ -874,11 +880,12 @@ namespace hazelcast {
                     std::vector<std::pair<serialization::pimpl::Data, serialization::pimpl::Data> > dataResult = proxy::IMapImpl::entrySetData(
                             predicate);
                     size_t size = dataResult.size();
-                    std::vector<std::pair<K, V> > entries(size);
+                    std::vector<std::pair<K, V> > entries;
+                    entries.reserve(size);
                     for (size_t i = 0; i < size; ++i) {
                         std::unique_ptr<K> key = toObject<K>(dataResult[i].first);
                         std::unique_ptr<V> value = toObject<V>(dataResult[i].second);
-                        entries[i] = std::make_pair(*key, *value);
+                        entries.emplace_back(*key, *value);
                     }
                     return entries;
                 }

--- a/hazelcast/src/hazelcast/client/serialization.cpp
+++ b/hazelcast/src/hazelcast/client/serialization.cpp
@@ -60,6 +60,10 @@ namespace hazelcast {
             return jsonString == rhs.jsonString;
         }
 
+        bool HazelcastJsonValue::operator<(const HazelcastJsonValue &rhs) const {
+          return jsonString < rhs.jsonString;
+        }
+
         bool HazelcastJsonValue::operator!=(const HazelcastJsonValue &rhs) const {
             return !(rhs == *this);
         }

--- a/hazelcast/test/src/HazelcastTests5.cpp
+++ b/hazelcast/test/src/HazelcastTests5.cpp
@@ -2391,6 +2391,7 @@ namespace hazelcast {
                 static const char *intMapName;
                 static const char *employeesMapName;
                 static const char *imapName;
+                static const char *jsonMapName;
                 static const std::string ONE_SECOND_MAP_NAME;
 
                 MapClientConfig() {
@@ -2403,6 +2404,7 @@ namespace hazelcast {
 
             const char *MapClientConfig::intMapName = "IntMap";
             const char *MapClientConfig::employeesMapName = "EmployeesMap";
+            const char *MapClientConfig::jsonMapName = "jsonMapTest";
             const char *MapClientConfig::imapName = "clientMapTest";
             const std::string MapClientConfig::ONE_SECOND_MAP_NAME = "OneSecondTtlMap";
 
@@ -2450,7 +2452,8 @@ namespace hazelcast {
                 ClientMapTest() : client(HazelcastClient(GetParam())),
                                   imap(client.getMap<std::string, std::string>(MapClientConfig::imapName)),
                                   intMap(client.getMap<int, int>(MapClientConfig::intMapName)),
-                                  employees(client.getMap<int, Employee>(MapClientConfig::employeesMapName)) {
+                                  employees(client.getMap<int, Employee>(MapClientConfig::employeesMapName)),
+                                  jsonMap(client.getMap<std::string, HazelcastJsonValue>(MapClientConfig::jsonMapName)) {
                 }
 
                 static void SetUpTestCase() {
@@ -2738,6 +2741,7 @@ namespace hazelcast {
                 IMap<std::string, std::string> imap;
                 IMap<int, int> intMap;
                 IMap<int, Employee> employees;
+                IMap<std::string, HazelcastJsonValue> jsonMap;
 
                 static HazelcastServer *instance;
                 static HazelcastServer *instance2;
@@ -3291,6 +3295,26 @@ namespace hazelcast {
                 t2.join();
                 ASSERT_FALSE(imap.isLocked("key1"));
 
+            }
+
+            TEST_P(ClientMapTest, testJsonValues) {
+                const int numItems = 5;
+                for (int i = 0; i < numItems; ++i) {
+                  jsonMap.put("key_" + std::to_string(i), HazelcastJsonValue("{ \"value\"=\"value_" + std::to_string(i) + "\"}"));
+                }
+                std::vector<HazelcastJsonValue> values = jsonMap.values();
+                ASSERT_EQ(numItems, (int) values.size());
+            }
+
+            TEST_P(ClientMapTest, testJsonValuesWithPagingPredicate) {
+              const int numItems = 5;
+              const int predSize = 3;
+              for (int i = 0; i < numItems; ++i) {
+                jsonMap.put("key_" + std::to_string(i), HazelcastJsonValue("{ \"value\"=\"value_" + std::to_string(i) + "\"}"));
+              }
+              query::PagingPredicate<std::string, HazelcastJsonValue> predicate((size_t) predSize);
+              std::vector<HazelcastJsonValue> values = jsonMap.values(predicate);
+              ASSERT_EQ(predSize, (int) values.size());
             }
 
             TEST_P(ClientMapTest, testValues) {

